### PR TITLE
Test Kotlin 2.1.0-Beta2 in CI

### PR DESCRIPTION
**Pull request by drewhamilton**
Tests compatibility of current Poko main branch with the new Kotlin beta version.

### CODE REVIEW

File: .github/workflows/ci.yml [hunk 1]
```diff
@@ -62,7 +62,7 @@ jobs:
     needs: build
     strategy:
       matrix:
-        poko_sample_kotlin_version: [ 2.0.20, ~ ]
+        poko_sample_kotlin_version: [ 2.0.20, ~, 2.1.0-Beta1 ]
```
Comment (JakeWharton) [inline .github/workflows/ci.yml:65]:
```suggestion
        poko_sample_kotlin_version: [ 2.0.20, ~, 2.1.0-Beta2 ]
```
Out yesterday

Comment (drewhamilton) [reply .github/workflows/ci.yml:65]: Yeah, just wanted to finish up old work before piling on new work (if there is any from the Beta2 bump).

**Comment by drewhamilton** [comment]
```
java.lang.NoSuchMethodError: 'org.gradle.api.provider.Property org.jetbrains.kotlin.gradle.dsl.KotlinCommonCompilerToolOptions.getExtraWarnings()'
```

This is thrown with 2.1.0-Beta2. I think this means we just won't be able to make 2.1 and 2.0 compatible, much like 2.0 and 1.9. Unless there's some Gradle magic we can do?

**Comment by JakeWharton** [comment]
That stack trace is weird. Isn't it KGP calling into itself that is causing the exception? Are there two KGPs on the Gradle classpath or something?

**Comment by JakeWharton** [comment]
This code:

https://github.com/drewhamilton/Poko/blob/4e710feff6d2d35633bb27865b33cfca34864ad5/sample/settings.gradle.kts#L76-L84

Needs to also live here:

https://github.com/drewhamilton/Poko/blob/4e710feff6d2d35633bb27865b33cfca34864ad5/sample/buildSrc/settings.gradle.kts#L3-L5

Otherwise the `buildSrc` uses the enclosing project's `libs.version.toml`'s KGP, whereas the sample itself uses the overridden KGP. This puts two copies onto the buildscript classpath which can cause a split-brain problem with classes from both being loaded at the same time producing otherwise impossible NSMEs and NCDFEs.

**Comment by JakeWharton** [comment]
I can send a PR for that later, if you don't get to it first.

**Comment by drewhamilton** [comment]
Ouch. Good find, fixed.

**Pull request merged**